ezz/jan/2023-01-20-각도기.java

```diff
@@ -0,0 +1,25 @@
+//문제 설명
+//각에서 0도 초과 90도 미만은 예각, 90도는 직각, 90도 초과 180도 미만은 둔각 180도는 평각으로 분류합니다. 
+//각 angle이 매개변수로 주어질 때 예각일 때 1, 직각일 때 2, 둔각일 때 3, 평각일 때 4를 return하도록 solution 함수를 완성해주세요.
+
+class Solution {
+    public int solution(int angle) {
+        int answer = 0;
+        //예각 : 1
+        if(angle < 90){
+            answer=1;
+        //직각 : 2    
+        }else if(angle==90){
+            answer=2;
+        //둔각 : 90
+        }else if(angle < 180){
+            answer=3; 
+        //평각 :4
+        }else if(angle == 180){
+            answer=4;
+        }                                
+     return answer;
```

Comment:
완벽.
if else문이 헷갈린다면 else를 사용하지 않고, 방법이 있다. 바로 if 문에서 return. 
(단 좋지 않다. answer값이 여러번 할당되고 바뀌는 경우)
바로 return 시키면 있다 장점. 하위 조건문으로 이동 X. 따라서 성능상 이점

```suggestion
        //예각 : 1
        if(angle < 90){
          return 1;  
        }
        //직각 : 2  
        if(angle==90){
            return 2;
        }
        //둔각 : 90
        if(angle < 180){
            return 3; 
        }
        //평각 :4
        if(angle == 180){
            return 4;
        }                                
```